ras_party/views/pending_survey_view.py

```diff
@@ -0,0 +1,194 @@
+    is_transfer = request.args.get('is_transfer', False)
+    if business_id and survey_id:
+        # this is just to validate that the business_id exists
+        get_business_by_id(business_id)
+        response = pending_survey_controller.get_users_enrolled_and_pending_survey_against_business_and_survey(
+            business_id, survey_id, is_transfer)
+        return make_response(jsonify(response), 200)
+    else:
+        raise BadRequest('Business id and Survey id is required for this request.')
+
+
+@pending_survey_view.route('/pending-surveys', methods=['POST'])
+def post_pending_surveys():
+    """
+     Creates new records for share survey
+     accepted payload example:
+     {  pending_shares: [{
+            "business_id": "business_id"
+            "survey_id": "survey_id",
+            "email_address": "email_address",
+            "shared_by": "respondent_id"
+        },
+        {
+            "business_id": "business_id":
+            "survey_id": "survey_id",
+            "email_address": "email_address",
+            "shared_by": "respondent_id"
+        }]
+     }
+     Creates new records for transfer surveys
+     accepted payload example:
+     {  pending_transfers: [{
+            "business_id": "business_id"
+            "survey_id": "survey_id",
+            "email_address": "email_address",
+            "shared_by": "respondent_id"
+        },
+        {
+            "business_id": "business_id":
+            "survey_id": "survey_id",
+            "email_address": "email_address",
+            "shared_by": "respondent_id"
+        }]
+     }
+    """
+    payload = request.get_json() or {}
+    if 'pending_shares' not in payload and 'pending_transfers' not in payload:
+        raise BadRequest('Payload Invalid - Pending survey key missing')
+    if 'pending_shares' in payload:
+        is_transfer = False
+        pending_surveys = payload['pending_shares']
+        existing_user_email_template = 'share_survey_access_existing_account'
+        new_user_email_template = 'share_survey_access_new_account'
+    else:
+        is_transfer = True
+        pending_surveys = payload['pending_transfers']
+        existing_user_email_template = 'transfer_survey_access_existing_account'
+        new_user_email_template = 'transfer_survey_access_new_account'
+
+    business_list = []
+    if len(pending_surveys) == 0:
+        raise BadRequest('Payload Invalid - pending_surveys list is empty')
+    for survey in pending_surveys:
+        # Validation, curation and checks
+        expected = ('email_address', 'survey_id', 'business_id', 'shared_by')
+        v = Validator(Exists(*expected))
+        if not v.validate(survey):
+            logger.debug(v.errors)
+            raise BadRequest(v.errors)
+    respondent = get_respondent_by_id(pending_surveys[0]['shared_by'])
+    try:
+        get_respondent_by_email(pending_surveys[0]['email_address'])
+        email_template = existing_user_email_template
```

Comment:
I don't think it was worth assigning variables to these; I'd just put `email_template = 'transfer_survey_access_existing_account` here (and likewise below)

Reply:
This will change based on where the flow is for transfer or share survey, hence assigned a variable